Shared/Supporting Files/Views/CategoryGridView.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct CategoryGridView: View {
+    /// All samples that will be shown in the categories.
+    private let samples: [Sample]
+    
+    /// The different sample categories generated from the samples list.
+    private let categories: [String]
+    
+    /// The columns for the grid.
+    private let columns = [
+        GridItem(.flexible()),
+        GridItem(.flexible())
+    ]
+    
+    init(samples: [Sample]) {
+        self.samples = samples
+        let categoriesSet = Set(samples.map(\.category))
+        categories = categoriesSet.sorted()
+    }
+    
+    var body: some View {
+        ScrollView {
+            LazyVGrid(columns: columns) {
+                ForEach(categories, id: \.self) { category in
+                    NavigationLink {
+                        SampleListView(samples: samples.filter { $0.category == category })
+                            .navigationTitle(category)
+                    } label: {
+                        CategoryTitleView(category: category)
+                    }
+                    .clipShape(RoundedRectangle(cornerRadius: 15))
+                    .contentShape(RoundedRectangle(cornerRadius: 30))
+                }
```

Comment:
```suggestion
                    .contentShape(RoundedRectangle(cornerRadius: 30))
                    .buttonStyle(.plain)
                }
```

Mac Catalyst before:
<img width="327" alt="Screenshot 2023-07-17 at 18 15 37" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/0d6b49d2-ac81-42c8-8193-f6b135981add">

After:              
<img width="354" alt="Screenshot 2023-07-17 at 18 18 04" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/a288e5ee-ffee-4290-bdd7-22442d63a773">